Remotion/Web/Core/UI/Controls/CommandInfo.cs

```diff
@@ -157,12 +157,27 @@ private void AddDiagnosticMetadataAttributes (HtmlTextWriter writer)
 
     private bool IsTriggeringNavigation ()
     {
-      return _href != "#";
+      if (_href == null)
+        return false;
+      else if (_href.StartsWith("#"))
+        return false;
+      else if (_onClick == null)
+        return true;
+      else if (_onClick.Contains("__doPostBack"))
+        return false;
+      else
+        return true;
     }
 
     private bool IsTriggeringPostBack ()
     {
-      return !IsTriggeringNavigation() && _onClick!.Contains("__doPostBack"); // TODO RM-8104: Guard _onClick
+      if (IsTriggeringNavigation())
+        return false;
+      else if (_onClick!.Contains("__doPostBack"))
+        return true;
+      else
+        return false;
+      // TODO RM-8104: Guard _onClick
     }
   }
 }
```

Comment:
We also need one more unit test:

```


    [Test]
    public void AddDiagnosticMetadataAttributes_FromCreateForLink_WithDoPostBackAndUrl ()
    {
      var commandInfo = CommandInfo.CreateForLink("TheTitle", null, "TheUrl", "TheTarget", "FrontGarbage__doPostBackBackGarbage");

      var stringWriter = new StringWriter();
      var htmlTextWriter = new HtmlTextWriter(stringWriter);
      commandInfo.AddAttributesToRender(htmlTextWriter, RenderingFeatures.WithDiagnosticMetadata);

      htmlTextWriter.RenderBeginTag(HtmlTextWriterTag.A);
      htmlTextWriter.RenderEndTag();

      var result = stringWriter.ToString();

      Assert.That(result, Does.Contain(DiagnosticMetadataAttributes.ControlType + "=\"Command\""));
      Assert.That(result, Does.Contain(DiagnosticMetadataAttributes.TriggersPostBack + "=\"true\""));
      Assert.That(result, Does.Contain(DiagnosticMetadataAttributes.TriggersNavigation + "=\"false\""));
    }
```

Please add it after `AddDiagnosticMetadataAttributes_FromCreateForLink_WithDoPostBack`.
The various tests for null cannot actually be tested thanks to guard conditions outside the scope of the private methods.